src/xunit.performance.api/XunitBenchmark.cs

```diff
@@ -90,16 +91,24 @@ public static (IEnumerable<ProviderInfo> providers, IEnumerable<PerformanceTestM
 
                     providers = ProviderInfo.Merge(providers);
 
-                    // Inject implicit pmc counters
-                    foreach (var test in testMessageVisitor.Tests)
+                    // Create the list of default metrics
+                    IEnumerable<PerformanceMetricInfo> defaultMetricInfos =
+                        from pmcMetricInfo in PerformanceMetricInfos
+                        where pmcMetricInfo.IsValidPmc
+                        select pmcMetricInfo;
+
+                    if (AllocatedBytesForCurrentThread.IsAvailable)
+                        defaultMetricInfos = defaultMetricInfos.Concat(new[] { new GCAllocatedBytesForCurrentThreadMetric() });
+                    else
                     {
-                        var metrics = new List<PerformanceMetricInfo>(test.Metrics);
-                        foreach (var performanceMetricInfo in PerformanceMetricInfos)
-                            if (performanceMetricInfo.IsValidPmc)
-                                metrics.Add(performanceMetricInfo);
-                        test.Metrics = metrics;
+                        WriteWarningLine(AllocatedBytesForCurrentThread.NoAvailabilityReason);
+                        WriteWarningLine("The 'Allocations In Current Thread' metric will not be collected.");
```

Comment:
Name change here as well... I think you should probably get the name from the metric here so that it's always consistent.